src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDaoMockTest.java

```diff
@@ -392,7 +412,148 @@ public void deletePermenantlyLogicallyDeletedSchedulePlanWorks() {
         
         dao.deleteSchedulePlanPermanently(TEST_STUDY, schedulePlan.getGuid());
         
-        verify(mapper).delete(any());
+        verify(mockMapper).delete(any());
+    }
+    
+    @Test
+    public void createSchedulePlan() {
+        when(dao.generateGuid()).thenReturn(GUID);
+        
+        // This creates a criteria schedule plan which is the most complicated to persist.
+        SchedulePlan plan = constructSchedulePlan();
+        plan.setStudyKey(null); // not allowed, should be set to study argument
+        plan.setDeleted(true); // not allowed, should be set to false
+        plan.setVersion(1L); // not allowed, should be set to null
+        Criteria criteria = ((CriteriaScheduleStrategy)plan.getStrategy()).getScheduleCriteria().get(0).getCriteria();
+        criteria.setKey(null); // verify this is set by the dao
+        
+        SchedulePlan returned = dao.createSchedulePlan(TEST_STUDY, plan);
+        
+        // Verify the criteria were persisted
+        verify(mockCriteriaDao).createOrUpdateCriteria(criteriaCaptor.capture());
+        assertSame(criteriaCaptor.getValue(), criteria);
+        assertEquals(criteriaCaptor.getValue().getKey(), SCHEDULE_CRITERIA_KEY);
+        
+        verify(mockMapper).save(schedulePlanCaptor.capture());
+        SchedulePlan persisted = schedulePlanCaptor.getValue();
+
+        assertEquals(persisted.getStudyKey(), TEST_STUDY_IDENTIFIER);
+        assertEquals(persisted.getGuid(), GUID);
+        assertEquals(persisted.getModifiedOn(), TIMESTAMP.getMillis());
+        assertFalse(persisted.isDeleted());
+        assertNull(persisted.getVersion());
+        
+        // verify as well that this is all returned to the caller
+        assertSame(returned, plan);
+    }
+    
+    @Test
+    public void updateSchedulePlan() {
+        schedulePlan.setStudyKey(null); // this will be set
```

Comment:
It looks like this test is backwards. We nullify the studyKey in the schedulePlan returned by get, but the schedule plan we pass in to the update() call has the studyKey, so we don't ever actually test that update sets the studyKey.

Reply:
Updated